src/Tribe/Ignored_Events.php

```diff
@@ -0,0 +1,526 @@
+<?php
+// Don't load directly
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+if ( ! class_exists( 'Tribe__Events__Ignored_Events' ) ) {
+	/**
+	 * Ignored Events are fully powered by this class
+	 */
+	class Tribe__Events__Ignored_Events {
+		public static $ignored_status = 'tribe-ignored';
+
+		public static $legacy_deleted_post = 'deleted_event';
+
+		public static $legacy_origin = 'ical-importer';
+
+		/**
+		 * Static singleton variable
+		 *
+		 * @var self
+		 */
+		public static $instance;
+
+		/**
+		 * Static Singleton Factory Method
+		 *
+		 * @return self
+		 */
+		public static function instance() {
+			if ( ! isset( self::$instance ) ) {
+				self::$instance = new self;
+			}
+
+			return self::$instance;
+		}
+
+		private function __construct() {
+			add_action( 'init', array( $this, 'register_ignored_post_status' ) );
+			add_action( 'current_screen', array( $this, 'maybe_restore_events' ) );
+
+			add_filter( 'pre_delete_post', array( $this, 'pre_delete_event' ), 10, 3 );
+			add_action( 'trashed_post', array( $this, 'from_trash_to_ignored' ) );
+
+			add_filter( 'views_edit-' . Tribe__Events__Main::POSTTYPE, array( $this, 'add_ignored_view' ) );
+			add_filter( 'post_row_actions', array( $this, 'filter_actions' ), 10, 2 );
+
+			add_action( 'wp_ajax_tribe_convert_legacy_ignored_events', array( $this, 'ajax_convert_legacy_ignored_events' ) );
+
+			/**
+			 * Register Notices
+			 */
+			Tribe__Admin__Notices::instance()->register( 'legacy-ignored-events', array( $this, 'render_notice_legacy' ), 'dismiss=1&type=warning' );
+		}
+
+		public function render_notice_legacy() {
+			if ( ! Tribe__Admin__Helpers::instance()->is_post_type_screen( Tribe__Events__Main::POSTTYPE ) ) {
+				return false;
+			}
+
+			if ( empty( $_GET['post_status'] ) || $_GET['post_status'] !== self::$ignored_status ) {
+				return false;
+			}
+
+			if ( ! $this->has_legacy_deleted_posts() ) {
+				return false;
+			}
+
+			$html = '<p>' . '@TODO: Include a Cool message about why you are seen this Notice!' . '</p>';
```

Comment:
...If not, can we add it as a @todo comment above this line rather than having it within the string? At least in terms of my own local tools, it's more likely to be visible that way.